src/wallet/spend.cpp

```diff
@@ -1392,4 +1392,381 @@ bool FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& nFeeRet,
+    }
+
+    OutputType output_type = wallet.m_default_address_type;
+    if (opt_output_type) {
+        output_type = *opt_output_type;
+    } else {
+        // if no output type was specified, try to infer it from the source inputs
+        CTxDestination shared_destination = CNoDestination();
+        if (ExtractDestination(shared_script, shared_destination)) {
+            std::optional<OutputType> opt_shared_output_type = OutputTypeFromDestination(shared_destination);
+            if (opt_shared_output_type) {
+                output_type = *opt_shared_output_type;
+            }
+        }
+    }
+
+    const int num_recipients = NUM_DENIABILIZATION_OUTPUTS;
+    std::vector<CRecipient> recipients(num_recipients);
+    std::list<ReserveDestination> reservedests;
+    constexpr bool reservdest_internal = false; // TODO: Should this be "true" or "false". What does "internal" mean?
+    for (int recipient_index = 0; recipient_index < num_recipients; recipient_index++) {
+        bool lastRecipient = recipient_index == (num_recipients - 1);
+        if (!lastRecipient) {
+            // all recipients except for the last one,
+            // calculate a random range based on the remaining total random range and the number of remaining recipients
+            // then generate a random amount within that range
+            CAmount random_range = total_random_range / (num_recipients - recipient_index - 1);
+            CAmount random_amount = 0;
+            if (random_range > 0) {
+                random_amount = GetRand<CAmount>(random_range);
+                Assert(total_random_range >= random_amount);
+                total_random_range -= random_amount;
+            }
+            recipients[recipient_index].nAmount = min_post_split_amount + random_amount;
+        } else {
+            // the last recipient takes any leftover random amount and the estimated fee
+            recipients[recipient_index].nAmount = min_post_split_amount + total_random_range + estimated_tx_fee;
+        }
+
+        // the last recipient pays the tx fees
+        recipients[recipient_index].fSubtractFeeFromAmount = lastRecipient;
+
+        auto& reservedest = reservedests.emplace_back(&wallet, output_type);
+        CTxDestination dest;
+        auto op_dest = reservedest.GetReservedDestination(reservdest_internal);
+        if (!op_dest) {
+            return util::Error{_("Failed to reserve a new address.") + Untranslated(" ") + util::ErrorString(op_dest)};
+        }
+        dest = *op_dest;
+        recipients[recipient_index].dest = dest;
+        if (lastRecipient) {
+            // we don't expect to get change, but we provide the address to prevent CreateTransactionInternal from generating a change address
+            coin_control.destChange = dest;
+        }
+    }
+
+    CAmount recipient_amount = std::accumulate(recipients.cbegin(), recipients.cend(), CAmount{0}, [](CAmount sum, const CRecipient& recipient) { return sum + recipient.nAmount; });
```

Comment:
fold to several lines?

Reply:
Yes, indeed. I split the code into several lines in the latest commit.
Thanks!

Reply:
I hate that indent standard though, and yet I think now it's a bit better than before. Thanks